core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -137,28 +137,78 @@ def __init__(self,
+        ag_arg_prefix : str, default = "ag."
+            The key prefix to look for that indicates a parameter is intended for params_aux.
+            If None, this logic is skipped.
+            If a key starts with this prefix, it is popped from params and added to params_aux with the prefix removed.
+            For example:
+                input:  params={'ag.foo': 2, 'abc': 7}, params_aux={'bar': 3}, and ag_arg_prefix='.ag',
+                output: params={'abc': 7}, params_aux={'bar': 3, 'foo': 2}
+            In cases where the key is specified multiple times, the value of the key with the prefix will always take priority.
+            A warning will be logged if a key is present multiple times.
+            For example, given the most complex scenario:
+                input:  params={'ag.foo': 1, 'foo': 2, 'ag_args_fit': {'ag.foo': 3, 'foo': 4}}
+                output: params={'foo': 2}, params_aux={'foo': 1}
+
+        Returns
+        -------
+        params, params_aux : (Dict[str, Any], Dict[str, Any])
+            params will contain the native model hyperparameters
+            params_aux will contain special auxiliary hyperparameters
+        """
+        params = copy.deepcopy(params) if params is not None else dict()
+        assert isinstance(params, dict), f"Invalid dtype of params! Expected dict, but got {type(params)}"
+        params_aux = params.pop(ag_args_fit, dict())
+        if params_aux is None:
+            params_aux = dict()
+        assert isinstance(params_aux, dict), f"Invalid dtype of params_aux! Expected dict, but got {type(params_aux)}"
+        if ag_arg_prefix is not None:
+            param_aux_keys = list(params_aux.keys())
+            for k in param_aux_keys:
+                if isinstance(k, str) and k.startswith(ag_arg_prefix):
+                    k_no_prefix = k[len(ag_arg_prefix):]
+                    if k_no_prefix in params_aux:
+                        logger.warning(f'Warning: {cls.__name__} hyperparameter "{k}" is present '
+                                       f'in `ag_args_fit` as both "{k}" and "{k_no_prefix}". '
+                                       f'Will use "{k}" and ignore "{k_no_prefix}".')
+                    params_aux[k_no_prefix] = params_aux[k]
+                    params_aux.pop(k)
+            param_keys = list(params.keys())
+            for k in param_keys:
+                if isinstance(k, str) and k.startswith(ag_arg_prefix):
+                    k_no_prefix = k[len(ag_arg_prefix):]
+                    if k_no_prefix in params_aux:
+                        logger.warning(f'Warning: {cls.__name__} hyperparameter "{k}" is present '
+                                       f'in both `ag_args_fit` and `hyperparameters`. '
+                                       f'Will use `hyperparameters` value.')
+                    params_aux[k_no_prefix] = params[k]
+                    params.pop(k)
```

Comment:
this is effectively a repeating code block as above, just on different nesting level -> do we want to generalize this function to multi-level nested construct?

Reply:
Given that there are meaningful differences between the code blocks and very different log statements, I think it is ok to keep as is. If you'd like to propose a code refactor of the logic, I'd be happy to review.